File: bddtests/src/main/kotlin/mocking/tpp/models/ListRepeatMedicationReply.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlRootElement
import javax.xml.bind.annotation.XmlElement

@XmlRootElement(name = "ListRepeatMedicationReply")
@XmlAccessorType(XmlAccessType.FIELD)
data class ListRepeatMedicationReply (
        @XmlAttribute var patientId: String = "default patientId",
        @XmlAttribute var onlineUserId: String = "default onlineUserId",
        @XmlAttribute var uuid: String = "default uuid",
        @XmlElement var Medication: MutableCollection<Medication> = mutableListOf()
)

File: backendworker/NHSOnline.Backend.Worker/GpSystems/Suppliers/Tpp/Models/Prescriptions/ListRepeatMedicationReply.cs
using System;
using System.Collections.Generic;
using System.Xml.Serialization;

namespace NHSOnline.Backend.Worker.GpSystems.Suppliers.Tpp.Models.Prescriptions
{
    [Serializable]
    public class ListRepeatMedicationReply
    {
        [XmlAttribute("patientId")]
        public string PatientId { get; set; }
        
        [XmlAttribute("onlineUserId")]
        public string OnlineUserId { get; set; }
        
        [XmlElement("Medication", typeof(Medication))]
        public List<Medication> Medications { get; set; }
        
        [XmlAttribute("uuid")]
        public Guid Uuid { get; set; }       
    }
}